csrc/tensor_metadata.cpp

```diff
@@ -315,18 +318,30 @@ inferAndValidateAllocationSizesAndStrides(
   allocation_sizes.reserve(alloc.size());
   allocation_strides.reserve(alloc.size());
   for (IterDomain* id : alloc | TensorDomain::kNoReductions) {
+    auto it = active_ids.find(id);
+    NVF_ERROR(it != active_ids.end(), "Allocation domain is not complete");
```

Comment:
nitpick: this seems to be a behavior change, such that device dim ID on allocation domain needs to be on logical domain, otherwise it asserts here.

Reply:
This is not a nitpick, it causes real bugs. Thanks for finding!